### PR TITLE
Delete message controller from 9103-replace-fim-db-dbsync

### LINKED ISSUE #14133: FIM DB Development - Remove RSync Message Controller
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| [9103-replace-fim-db-dbsync](https://github.com/wazuh/wazuh/tree/9103-replace-fim-db-dbsync) | FIM | All | Sources | All|

Hi team!

After the changes introduced in https://github.com/wazuh/wazuh/pull/14105, FIM DB is not going to configure any more the option `min_interval`. This means that some of the changes introduced in https://github.com/wazuh/wazuh/pull/13558 could be removed.

This issue aims to address any possible feature added that is not going to be used.

### PR DESCRIPTION
**Pull request by FrancoRivero**
|EPIC|Component|
|---|---|
| https://github.com/wazuh/wazuh/issues/9103 | FIM |

|Related issue|
|---|
|Closes #14133|
### Description
Hello team, this pull-request is to remove the message controller as a feature of DBSync in order to have compatibility with master and have less merge conflicts.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X
- [x] Unit tests passed
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [x] cppcheck (static code analysis)
  - [x] Coverage (unit tests coverage code)
  - [x] Sformat (code style)
  - [x] Scan build

### CODE REVIEW

**Review by jotacarma90 — APPROVED** [review]
LGTM

File: src/shared_modules/rsync/include/rsync.hpp [hunk 1]
```diff
@@ -264,14 +264,6 @@ class EXPORTED RegisterConfiguration final : public Configuration<RegisterConfig
         */
         RegisterConfiguration & decoderType(const std::string& decoderType);
 
-        /**
-         * @brief Set the minimal sync interval time for the message controller.
-         *
-         * @param intervalTime Time in seconds to be set.
-         *
-         */
-        RegisterConfiguration & minimalSyncIntervalTime(const uint32_t intervalTime);
```
Comment (Dwordcito) [inline src/shared_modules/rsync/include/rsync.hpp:273]: Remove, line 346

StartSyncConfiguration & syncOnDemand(const bool syncOnDemand);

File: src/shared_modules/rsync/src/rsync.cpp [hunk 2]
```diff
@@ -262,6 +262,13 @@ void RemoteSync::registerSyncID(const std::string&    messageHeaderID,
                                 const nlohmann::json& syncConfiguration,
                                 SyncCallbackData      callbackData)
 {
+    const auto callbackWrapper
+    {
+        [callbackData](const std::string & payload)
+        {
+            callbackData(payload);
+        }
+    };
```
Comment (Dwordcito) [inline src/shared_modules/rsync/src/rsync.cpp:271]: Remove this.
```suggestion
```

File: src/shared_modules/rsync/src/rsync.cpp [hunk 3]
```diff
@@ -336,12 +343,6 @@ RegisterConfiguration& RegisterConfiguration::rangeChecksum(QueryParameter& para
     return *this;
 }
 
-RegisterConfiguration& RegisterConfiguration::minimalSyncIntervalTime(const uint32_t parameter)
```
Comment (Dwordcito) [inline src/shared_modules/rsync/src/rsync.cpp:339]: remove 364-368 syncOnDemand

**Review by Dwordcito — CHANGES_REQUESTED** [review]

**Review by antoniomanuelfr — APPROVED** [review]

**Review by chemamartinez — APPROVED** [review]
LGTM

**Pull request merged**